wallet-api/src/Ledger/Types.hs

```diff
@@ -197,16 +197,18 @@ instance Serialise (Digest SHA256) where
       Just v  -> pure v
 
 instance ToJSON (Digest SHA256) where
-  toJSON = JSON.String . TE.decodeUtf8 . Base64.encode . Write.toStrictByteString . encode
+  toJSON = JSON.String . TE.decodeUtf8 . Base16.encode . Write.toStrictByteString . encode
 
 instance ToSchema (Digest SHA256) where
   declareNamedSchema _ = plain . paramSchemaToSchema $ (Proxy :: Proxy String)
 
 instance FromJSON (Digest SHA256) where
   parseJSON = withText "SHA256" $ \s -> do
     let ev = do
-          eun64 <- Base64.decode . TE.encodeUtf8 $ s
-          first show $ deserialiseOrFail $ BSL.fromStrict eun64
+          let (eun16, rest) = Base16.decode . TE.encodeUtf8 $ s
+          if BSS.null rest
+          then first show $ deserialiseOrFail $ BSL.fromStrict eun16
+          else fail "failed to decode base16"
```

Comment:
The duplication of this code is annoying.

Reply:
I agree, there must be a place we can put this

Reply:
OK